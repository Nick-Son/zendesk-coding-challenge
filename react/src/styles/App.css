.App {
  text-align: center;
  background-size: cover;
  height: 100vh;
}

.header {
  background: linear-gradient(to right, #FDF4EA , #FCCDBF);
  color: #03363D;
  display: flex;
  padding: 5px;
}

.header__logo {
  display: block;
  height: 50px;
}

.header__title {
  font-size: 1.5em;
  font-family: Arial, Helvetica, sans-serif;
  margin: 10px auto;
  padding-top: 5px;
  padding-left: 50px;
  width: 500px;
}

.ticket-list {
  border-radius: 10px;
  color: #03363D;
  background: white;
  margin: 20px auto;
  padding: 10px;
  width: 600px;
  box-shadow: 0px 3px 15px rgba(33, 47, 128, 0.2);
}

.ticket-list-item {
  border: 1px solid lightgray;
  border-radius: 5px;
  cursor: pointer;
  margin: 5px;
}

.ticket-list-item:hover {
  background: #FDF4EA;
  transition: .9s;
}

.ticket-list-item__header {
  text-align: left;
  padding-left: 10px;
}

.date {
  color: grey;
}

.text-left {
  text-align: left;
  padding-left: 10px;
}

.right {
  float: right;
  padding-right: 10px;
}

.error {
  background: white;
  border: 2px solid salmon;
  border-radius: 5px;
  width: 400px;
  padding: 5px;
  margin: 40px auto;
}

.error__message {
  color: red;
}

.page-selector {
  max-width: 500px;
  list-style: none;
  display: flex;
  justify-content: center;
  margin: 1rem auto;
  overflow: hidden;
  padding: 5px;
}

.page-selector__link {
  border: .5px solid lightgrey;
  padding: 10px;
  user-select: none;
  cursor: pointer;
}

.current-page {
  background: #FCCDBF;
  color: white;
}

.page-selector__link:hover {
  background: #FDF4EA;
  color: white;
}

.page-selector__link:first-child {
  border-top-left-radius: 10px;
  border-bottom-left-radius: 10px;
}

.page-selector__link:last-child {
  border-top-right-radius: 10px;
  border-bottom-right-radius: 10px;
}

/* Modal */
.modal {
  background: white;
  border: .5px dotted lightgray;
  border-radius: 10px;
  width: 700px;
  outline: none;
  padding: 10px;
  text-align: center;
  margin-top: -200px;
  box-shadow: 0px 3px 15px rgba(33, 47, 128, 0.2);
}

.ReactModalPortal > div {
  opacity: 0;
}

.ReactModalPortal .ReactModal__Overlay {
  display: flex;
  justify-content: center;
  align-items: center;
  transition: opacity 200ms ease-in-out;
}

.ReactModalPortal .ReactModal__Overlay--after-open {
  opacity: 1;
}

.ReactModalPortal .ReactModal__Overlay--before-close {
  opacity: 0;
}

.button {
  background: #FCCDBF;
  border: none;
  padding: 10px;
  font-weight: 400;
  font-size: 1.1rem;
  border-radius: 10px;
  margin: 0 auto;
  width: 200px;
}

.dotted-line {
  border-bottom: .5px solid lightgray;
  padding-bottom: 10px;
}

.ticket-modal__description {
  padding: 15px;
  text-align: left;
  margin: 10px;
  border-bottom: 0.5px dotted lightgray;
}

.loader {
  margin: 200px auto;
}
