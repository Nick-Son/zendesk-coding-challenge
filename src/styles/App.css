.App {
  text-align: center;
  background-image: url("../assets/bg.png");
  background-size: cover;
  height: 100vh;
  width: 100vw;
}

.App-logo {
  display: block;
  height: 50px;
}

.App-header {
  /* background-color: #FDF4EA; */
  /* background-color: #FCCDBF; */
  background: linear-gradient(to right, #FDF4EA , #FCCDBF);
  color: #03363D;
  display: flex;
  padding: 5px;
}

.App-title {
  font-size: 1.5em;
  margin: 10px auto;
  padding-right: 100px;
  width: 500px;
}

.ticket-list {
  border-radius: 3px;
  border: 1px solid red;
  color: #03363D;
  background: white;
  margin: 20px auto;
  width: 600px;
}