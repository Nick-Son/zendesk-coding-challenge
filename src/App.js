import React, { Component } from 'react';
import logo from './assets/zendesk-medium-white.svg';
import './styles/App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h3 className="App-title">Zendesk Ticket Viewer</h3>
        </header>
        <p className="App-intro">
          Display TicketList component
        </p>
      </div>
    );
  }
}

export default App;
