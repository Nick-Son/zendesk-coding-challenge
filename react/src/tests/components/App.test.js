import React from 'react';
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJSON from 'enzyme-to-json'
import App from '../../App';
// import ticketData from '../fixtures/ticketData'

Enzyme.configure({
  adapter: new Adapter()
})

test('Should render App correctly', () => {
  const wrapper = shallow(<App />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
