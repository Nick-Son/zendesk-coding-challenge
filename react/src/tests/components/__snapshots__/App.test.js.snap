// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render App correctly 1`] = `
<div
  className="App"
>
  <Header />
  <div>
    <img
      alt="loading"
      className="loader"
      src="loader.svg"
    />
    <TicketModal
      onClearSelectedTicket={[Function]}
    />
  </div>
</div>
`;
